### DISCUSSION TITLE
How to export an array with a default value to OpenAPI

### DISCUSSION THREAD

**Discussion by khansgithub** [cue-lang/cue#2475] [Q&A] [answered]
Hi!

cue:
```
#foo: ["foobar", ...string]
```
This cue definition is converted to the below OpenAPI spec. However the `items` definition seems to not follow the OpenAPI spec. 

openapi:
```
...
"schemas": {
    "foo": {
        "type": "array",
        "minItems": 1,
        "items": [
            {
                "type": "string",
                "enum": [
                    "foobar"
                ]
            }
        ],
        "default": [
            "foobar"
        ],
        "additionalItems": {
            "type": "string"
        }
    }
}
...
```

validation errors:
> Structural error at components.schemas.foo
should NOT have additional properties
additionalProperty: additionalItems
Jump to line 8

> Structural error at components.schemas.foo.items
should be object
Jump to line 13

> Semantic error at components.schemas.foo.items
items must be an object
Jump to line 13

https://github.com/cue-lang/cue/discussions/1115
I saw this discussion about something similar (?) but unsure what to takeaway from it.

How can I go about exporting this to OpenAPI in a way that conforms with the spec :smile: ?

**Comment by myitcv** [comment]
@khansgithub did you intend to close this discussion?

**Reply by khansgithub** [reply, marked as answer]
Hi! Yes I did haha

A short while after I realised I might be asking the wrong questions, and eventually didn't require the OpenAPI export function :smile: So I thought to just close this.

**Reply by myitcv** [reply]
Ok great. Thanks for confirming!
